src/NexusMods.App.UI/Pages/Downloads/ViewModels/DownloadTaskViewModel.cs

```diff
@@ -1,129 +1,83 @@
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
 using NexusMods.App.UI.Resources;
+using NexusMods.MnemonicDB.Abstractions;
 using NexusMods.Networking.Downloaders.Interfaces;
-using NexusMods.Networking.Downloaders.Interfaces.Traits;
 using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
 
 namespace NexusMods.App.UI.Pages.Downloads.ViewModels;
 
 public class DownloadTaskViewModel : AViewModel<IDownloadTaskViewModel>, IDownloadTaskViewModel
 {
-    public IDownloadTask Task => _task;
     private readonly IDownloadTask _task;
 
-    public DownloadTaskViewModel(IDownloadTask task, bool initPreviousStates = true)
+    public DownloadTaskViewModel(IDownloadTask task)
     {
         _task = task;
-
-        // Initialize the previous states
-        if (!initPreviousStates)
-            return;
-
-        _previousName = Name;
-        _previousVersion = Version;
-        _previousGame = Game;
-        _previousStatus = Status;
-        _previousDownloadedBytes = DownloadedBytes;
-        _previousSizeBytes = SizeBytes;
-        _previousThroughput = Throughput;
-    }
-
-    public string Name => _task.FriendlyName;
-
-    public string Version
-    {
-        get
+        
+        this.WhenActivated(d =>
         {
-             if (_task is IHaveDownloadVersion version)
-                 return version.Version;
-
-             return Language.DownloadTaskViewModel_Field_Unknown;
-        }
+            _task.WhenAnyValue(t => t.PersistentState.FriendlyName)
+                .OnUI()
+                .Select(t => t)
+                .BindTo(this, x => x.Name)
+                .DisposeWith(d);
+            
+            _task.WhenAnyValue(t => t.PersistentState.Version)
+                .OnUI()
+                .BindTo(this, x => x.Version)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.PersistentState.Status)
+                .OnUI()
+                .BindTo(this, x => x.Status)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.Downloaded)
+                .OnUI()
+                .Select(s => s.Value)
+                .BindTo(this, x => x.DownloadedBytes)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.PersistentState.Size)
+                .OnUI()
+                .Select(s => s.Value)
+                .BindTo(this, x => x.SizeBytes)
+                .DisposeWith(d);
+            
+            _task.WhenAnyValue(t => t.PersistentState.GameDomain)
+                .OnUI()
+                .Select(g => g.ToString())
+                .BindTo(this, x => x.Game)
+                .DisposeWith(d);
+
+            _task.WhenAnyValue(t => t.Bandwidth)
+                .OnUI()
+                .Select(b => b.Value)
+                .BindTo(this, x => x.Throughput)
+                .DisposeWith(d);
```

Comment:
If we know that one of these always updates, it would be more performant to just a have a single subscription that updates all the values instead of many subscriptions. But this should work for now